### ISSUE TITLE
[R-package] Export callback functions

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#2479]
See [this comment](https://github.com/microsoft/LightGBM/pull/2443#issuecomment-536321691) and [this comment thread](https://github.com/microsoft/LightGBM/pull/2443#issuecomment-536314106) from #2443 . Currently the functions in `callback.R` are not exported in the R package. This means that the use of `callbacks` keyword argument in `lgb.train()` is unclear.

To close this issue, submit a PR with the following features

* [ ] Roxygen documentation on relevant functions in `R-package/callbacks.R`, including `@export` tag. Consider also using `@family` tag to auto-generate `see-also` links.
* [ ] regenerated `.Rd` documentation (see [this comment](https://github.com/microsoft/LightGBM/pull/2437#issuecomment-536244608) for how to do this). This should also regenerate `R-package/NAMESPACE`.
* [ ] updated documentation in any other functions that expose `callbacks`, linking to the relevant documentation

**Comment by jameslamb** [comment]
Closing this in favor of being in #2302 . We keep all feature requests in #2302 . If you'd like to pick this issue up and submit a PR, please do and it can be re-opened.

**Comment by jameslamb** [comment]
PR #5018 has been closed. Since there's no active work going on for this feature, I'm going to close this issue.

Anyone interested in this feature is welcome to contribute it! If you'd like to contribute this, comment here and the issue can be re-opened. Maintainers will be happy to help you.

**Comment by StrikerRUS** [comment]
@jameslamb I see you
>self-assigned this on Jan 3

Are you still working on this feature request?

**Comment by jameslamb** [comment]
Yes, thanks for the reminder, I'd like to get to this within the next release cycle. Not having it is blocking #6172. Let's leave it open please.
